addons/mrp/tests/test_byproduct.py

```diff
@@ -473,3 +473,22 @@ def test_03_check_byproducts_update(self):
         mo.button_mark_done()
         self.assertEqual(len(mo.move_byproduct_ids.move_line_ids), 2)
         self.assertEqual(mo.move_byproduct_ids.product_id, mo.move_byproduct_ids.move_line_ids.product_id)
+
+    def test_byproduct_qty_update(self):
+        """
+        Test that byproduct quantity is updated to the quantity set on the Mo when the Mo is marked as done.ee
+        """
+        self.bom_byproduct.byproduct_ids.product_qty = 0.0
+        self.warehouse.manufacture_steps = 'pbm_sam'
+        mo = self.env["mrp.production"].create({
+            'product_id': self.product_a.id,
+            'product_qty': 1.0,
+            'bom_id': self.bom_byproduct.id,
+        })
+        mo.action_confirm()
+        mo.move_byproduct_ids.quantity = 1.0
+        mo.button_mark_done()
+        self.assertEqual(mo.state, 'done')
+        self.assertEqual(mo.picking_ids[1].state, 'assigned')
```

Comment:
I'm always a bit wary with checks like those. We've had a few nondeterministic errors in tests in the past due to messed up ordering. Although it should be fine in this case, I don't especially like having to rely on the record's `_order`.
Could you rather pre-filter directly what you're trying to check, so we can safely avoid this issue altogether? :slightly_smiling_face:

Reply:
Hello @clesgow, thank you for your review, here we had an orderset, so the order should not change randomly. But I agree with what you mentioned in your comment and I applied the necessary changes ;)